lib/index.d.ts

```diff
@@ -39,3 +45,76 @@ export class StateController<State> {
   /** @internal Unsubscribe the listener callback that was passed to subscribeUpdates */
   unsubscribeUpdates(listener: (props: Partial<State>) => void): void;
 }
+
+export class Component<State> extends WebComponent {
+    /**
+     * Defines the state of the component, including all the properties required for rendering.
+     */
+    state: State;
+    /**
+     * Defines standard component configuration.
+     */
+    config: Component.ComponentConfigOptions<State>;
+    /**
+     * Template helper functions defined in config object, and exposed to template code as $helpers.
+     * This getter uses the component's internal config cache.
+     */
+    helpers: object;
+    /**
+     * For use inside view templates, to create a child Panel component nested under this
+     * component, which will share its state object and update cycle.
+     */
+    child(tagName: string, config?: object): VNode;
+    /**
+     * Searches the component's Panel ancestors for the first component of the
+     * given type (HTML tag name).
+     */
+    findPanelParentByTagName(tagName: string): Component<any>;
+    /**
+     * Fetches a value from the component's configuration map (a combination of
+     * values supplied in the config() getter and defaults applied automatically).
+     */
+    getConfig(key: string): any;
+    /**
+     * Executes the route handler matching the given URL fragment, and updates
+     * the URL, as though the user had navigated explicitly to that address.
+     */
+    navigate(fragment: string, stateUpdate?: Partial<State>): void;
+    /**
+     * Sets a value in the component's configuration map after element
+     * initialization.
+     */
+    setConfig(key: string, val: any): void;
+    /**
+     * To be overridden by subclasses, defining conditional logic for whether
+     * a component should rerender its template given the state to be applied.
+     * In most cases this method can be left untouched, but can provide improved
+     * performance when dealing with very many DOM elements.
+     */
+    shouldUpdate(state: State): boolean;
+    /**
+     * Applies a state update, triggering a re-render check of the component as
+     * well as any other components sharing the same state. This is the primary
+     * means of updating the DOM in a Panel application.
+     */
+    update(stateUpdate?: Partial<State>): void;
```

Comment:
this should be `update(props?: Partial<State>)` right? is shouldUpdate a meaningful variable name?

Reply:
`stateUpdate` seems meaningful to me -- I tried to take names from the library where possible

Reply:
ha! I read that wrong, I read it as `shouldUpdate`. Fine as is